### PR TITLE
better async support

### LINKED ISSUE #61: async tests fail to find testing framework
```
[Test]
public async void AsyncTest()
{
    await Task.Delay(10);
    Approvals.Verify("Foo");
}
```

result in 

```
System.Exception : Approvals is not set up to use your test framework.
It currently supports [NUnit, MsTest, MbUnit, xUnit.net, xUnit.extensions, Machine.Specifications (MSpec)]
To add one use ApprovalTests.StackTraceParsers.StackTraceParser.AddParser() method to add implementation of ApprovalTests.StackTraceParsers.IStackTraceParser with support for your testing framework.
To learn how to implement one see http://blog.approvaltests.com/2012/01/creating-namers.html
   at ApprovalTests.StackTraceParsers.StackTraceParser.Parse(StackTrace stackTrace) in c:\code\ApprovalTests.Net\ApprovalTests\StackTraceParsers\StackTraceParser.cs: line 31
   at ApprovalTests.Namers.UnitTestFrameworkNamer..ctor() in c:\code\ApprovalTests.Net\ApprovalTests\Namers\UnitTestFrameworkNamer.cs: line 17
   at ApprovalTests.Approvals.<.cctor>b__c() in c:\code\ApprovalTests.Net\ApprovalTests\Approvals.cs: line 24
   at ApprovalTests.Approvals.GetDefaultNamer() in c:\code\ApprovalTests.Net\ApprovalTests\Approvals.cs: line 134
   at ApprovalTests.Approvals.Verify(IApprovalWriter writer) in c:\code\ApprovalTests.Net\ApprovalTests\Approvals.cs: line 97
   at ApprovalTests.Approvals.Verify(String text) in c:\code\ApprovalTests.Net\ApprovalTests\Approvals.cs: line 124
   at MyTests.<AsyncTests>d__4.MoveNext() in PowerShellRunnerTests.cs: line 28
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>b__0(Object state)
   at NUnit.Framework.AsyncSynchronizationContext.AsyncOperationQueue.InvokePendingOperations()
   at NUnit.Framework.AsyncSynchronizationContext.AsyncOperationQueue.InvokeAll()
   at NUnit.Framework.AsyncInvocationRegion.AsyncVoidInvocationRegion.WaitForPendingOperationsToComplete(Object invocationResult)
   at NUnit.Framework.AsyncInvocationRegion.AsyncVoidInvocationRegion.WaitForPendingOperationsToComplete(Object invocationResult)
   at NUnit.Core.NUnitAsyncTestMethod.RunTestMethod()
```

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #61

### CODE REVIEW

**Comment by SimonCropp** [comment]
i would actually like to escalate the ambiguous match but could no work out how to do it without breaking APIs. 

i aslo think the AttributeStackTraceParser is leaking implementation details but again I kept this change backwards compat.

I have been fairly cautious with the checking for async in the hope of avoiding incorrect matching. 

@isidore  Ping me if u want to pair on this?

**Comment by SimonCropp** [comment]
ok dont merge this yet I will move over to ` AsyncStateMachine(typeof(AsyncNamerTests.<TestSourcePath>d__0))]` instead

**Comment by CraigRichards** [comment]
This is exactly the issue we are facing. Any idea on when this commit might go through?

File: ApprovalTests.Tests/Namer/AsyncNamerTests.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace ApprovalTests.Tests.Namer
+{
+    [TestFixture]
+    public class AsyncNamerTests
+    {
+        [Test]
+        public async void TestSourcePath()
+        {
+            await Task.Delay(1);
```
Comment (JayBazuzi) [inline ApprovalTests.Tests/Namer/AsyncNamerTests.cs]: I think you want `await Task.Yield()` here.

**Pull request closed without merging**